docker.js

```diff
@@ -0,0 +1,187 @@
+const _ = require("lodash");
+
+function createVolumeConfig(path) {
+  if (!path) {
+    throw new Error("Path is required to create Volume Config.");
+  }
+  if (!_.isString(path)) {
+    throw new Error("Path parameter must be a string.");
+  }
+
+  const pathEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPointEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPoint = generateRandomTemporaryPath();
+
+  return {
+    path: {
+      environmentVariable: {
+        name: pathEnvironmentVariableName,
+        value: path,
+      },
+      value: `$${pathEnvironmentVariableName}`,
+    },
+    mountPoint: {
+      environmentVariable: {
+        name: mountPointEnvironmentVariableName,
+        value: mountPoint,
+      },
+      value: `$${mountPointEnvironmentVariableName}`,
+    },
+  };
+}
+
+function sanitizeCommand(command) {
+  if (!command || !_.isString(command)) {
+    throw new Error("Command parameter must be a string.");
+  }
+
+  return `sh -c ${JSON.stringify(command)}`;
+}
+
+function mergeVolumeConfigsEnvironmentVariables(volumeConfigs) {
+  if (!volumeConfigs || !_.isArray(volumeConfigs)) {
+    throw new Error("Volume Configs parameter must be an array of objects.");
+  }
+
+  const requiredByDocker = volumeConfigs.reduce((acc, curr) => {
+    assertVolumeConfigPropertiesExistence(curr, [
+      "mountPoint.environmentVariable.name",
+      "mountPoint.environmentVariable.value",
+    ]);
+
+    return {
+      ...acc,
+      [curr.mountPoint.environmentVariable.name]: curr.mountPoint.environmentVariable.value,
+    };
+  }, {});
+
+  const requiredByShell = volumeConfigs.reduce((acc, curr) => {
+    assertVolumeConfigPropertiesExistence(curr, [
+      "path.environmentVariable.name",
+      "path.environmentVariable.value",
+    ]);
+
+    return {
+      ...acc,
+      [curr.path.environmentVariable.name]: curr.path.environmentVariable.value,
+    };
+  }, requiredByDocker);
+
+  return {
+    namesRequiredByDocker: Object.keys(requiredByDocker),
+    requiredByShell,
+  };
+}
+
+function buildDockerCommand({
+  command,
+  image,
+  environmentVariables = [],
+  volumeConfigs = [],
+  workingDirectory = "",
+  user = "",
+  additionalArguments = [],
+}) {
+  if (!image) {
+    throw new Error("No Docker image provided.");
+  }
+  if (!command) {
+    throw new Error("No command provided for Docker container.");
+  }
+
+  if (
+    additionalArguments && (
+      !_.isArray(additionalArguments)
+      || _.some(additionalArguments, (additionalArgument) => !_.isString(additionalArgument))
+    )
+  ) {
+    throw new Error("Additional Arguments must be an array of strings.");
+  }
+
+  const environmentVariableArguments = buildEnvironmentVariableArguments(environmentVariables);
+  const volumeArguments = buildMountVolumeArguments(volumeConfigs);
+  const userArguments = user && ["--user", user];
```

Comment:
Basically here and in the condition below the same parameter is evaluated twice.

If you want to keep the arguments preparation and merging separate, you could set the arguments to empty value (eg. `[]`) if they're not present and then perform the merging on all arguments. In the end that's what preparation step is about - making sure the data is there and is organized.